packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/dataStreamSearch/workspaceSelector/workspaceSelector.tsx

```diff
@@ -16,20 +18,46 @@ export interface WorkspaceSelectorProps {
   control: Control<SearchFields>;
   client: IoTTwinMakerClient;
   OnGettingError: (isError: boolean) => void;
+  setValue: UseFormSetValue<SearchFields>;
 }
 
 export const WorkspaceSelector = ({
   control,
   client,
   OnGettingError,
+  setValue,
 }: WorkspaceSelectorProps) => {
   const { workspaces, status } = useWorkspaces({ client });
   const workspaceOptions = createWorkspaceOptions(workspaces);
 
+  const [storedWorkspace, setStoredWorkspace] =
+    useLocalStorage<OptionDefinition | null>('storedWorkspace', null);
+  if (storedWorkspace) setValue('workspace', storedWorkspace);
```

Comment:
I still do not know why we are setting the value on every render .... am i missing something ?

Reply:
@mnischay 
We changed our approach, now there is no need for doing this on each render, the code changes have been reduced to one file only. Please review. Functionality wise there is no change, the same videos in description can be referred.